src/sentry/sentry_metrics/consumers/indexer/batch.py

```diff
@@ -73,9 +73,15 @@ class InboundMessage(TypedDict):
 
 
 class IndexerBatch:
-    def __init__(self, use_case_id: UseCaseKey, outer_message: Message[MessageBatch]) -> None:
+    def __init__(
+        self,
+        use_case_id: UseCaseKey,
+        outer_message: Message[MessageBatch],
+        should_index_tag_values: bool,
```

Comment:
This is the most important change from the original bad PR (https://github.com/getsentry/sentry/pull/38837/files#diff-7697ef4fd4cf439ed7ad41b7d98c00ff7d24cde30953e12d6bcb877303bb942dR81). The option which was introduced in https://github.com/getsentry/sentry/pull/38758 was boolean based and has been setup in prod as `True`.

That means we either index tag values for all orgs or none.